### PR TITLE
Typescript improvements

### PR DESCRIPTION
**Pull request by italodr**
## Make typescript improvements in one component

We want to create a well-typed component to use as example in further typing enhancements.

### CODE REVIEW

File: frontend/components/App/FishbowlPreJoin/form-auth.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@
  * file that was distributed with this source code.
  */
 
+import React from 'react';
```
Comment (eddsaura) [inline frontend/components/App/FishbowlPreJoin/form-auth.tsx:10]: With nextjs we don't need to import it but, in case we need it, we should do it this way according this[ microsoft's guide](https://github.com/Microsoft/TypeScript-React-Conversion-Guide#typescript-react-conversion-guide). 

```suggestion
import * as React from 'react';
```

File: frontend/package.json [hunk 2]
```diff
@@ -52,6 +53,7 @@
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^12.1.2",
     "@types/jest": "^27.0.2",
+    "@types/jitsi-meet": "^2.0.2",
```
Comment (jordisala1991) [inline frontend/package.json:56]: This should be in another PR because it would be nice to take a look at those types. The package name is jitsi-meet and we are not using jitsi-meet. Our javascript library is lib-jitsi-meet

File: frontend/components/App/FishbowlPreJoin/index.tsx [hunk 3]
```diff
@@ -64,22 +63,22 @@ const FishbowlPreJoin: React.FC = () => {
     localTracks.current = [];
   };
 
-  const handleParentClick = event => {
-    if (event.target.id !== 'config-button') {
+  const handleParentClick = (event: React.MouseEvent<HTMLDivElement>): void => {
+    if ((event.target as HTMLDivElement).id !== 'config-button') {
```
Comment (jordisala1991) [inline frontend/components/App/FishbowlPreJoin/index.tsx:67]: this is redundant, since on the previous line you are already declaring it generic with HTmlDivElement

Comment (italodr) [reply frontend/components/App/FishbowlPreJoin/index.tsx:67]: It does not, as not al event targets are HTML Elements. Check [this thread](https://github.com/Microsoft/TypeScript/issues/29540) about this issue

File: frontend/components/App/FishbowlPreJoin/form-auth.tsx [hunk 4]
```diff
@@ -16,15 +17,15 @@ import { InputStyled } from 'ui/Form';
 import Button from 'ui/Button';
 import FormikForm from 'ui/Form';
 
-type TProps = {
+interface Props {
   name: string;
 };
 
-const AuthUser = ({ name }: TProps) => {
+const AuthUser: React.FC<Props> = ({ name }) => {
```
Comment (aerrasti) [inline frontend/components/App/FishbowlPreJoin/form-auth.tsx:24]: What do you think about using AuthUserForm and AuthUserProps?

File: frontend/components/App/FishbowlPreJoin/form.tsx [hunk 5]
```diff
@@ -59,14 +60,14 @@ const FormValidation = withFormik<FormProps, FormValues>({
   }
 })(Form);
 
-const Nickname = () => {
+const Nickname: React.FC = () => {
```
Comment (aerrasti) [inline frontend/components/App/FishbowlPreJoin/form.tsx:63]: And this one could be GuestForm

**Comment by aerrasti** [comment]
Maybe I'm not aware if we do have an agreement on the file naming convention.
- kebap case?
- Camel case?

My choice would be using Camel case that matches with the name of the component like : 

https://github.com/taniarascia/takenote/tree/master/src/client/components

any thoughts?

File: frontend/components/App/FishbowlPreJoin/form-auth.tsx (hunk 4 above)
Comment (eddsaura) [reply frontend/components/App/FishbowlPreJoin/form-auth.tsx:24]: Normally, while props are only in the component scope, they aren't needed to be specified like `AuthUserProps` .

There is another case of some forms/components where we need more than one interface per file then yes I'd do it this way.

I agree with AuthUserForm name.

File: frontend/components/App/FishbowlPreJoin/form.tsx (hunk 5 above)
Comment (eddsaura) [reply frontend/components/App/FishbowlPreJoin/form.tsx:63]: It is only one field, should we use NicknameField instead of `Form` word?

Comment (jordisala1991) [reply frontend/components/App/FishbowlPreJoin/form.tsx:63]: It is not only a field, it contains the whole logic about the form.
What is strange IMO is that on the form-auth there is only one component, and here we have one component for the "form" (without any logic) and another component.

Having 2 components on the same file seems not a great idea. (why can't we join them?)

**Comment by italodr** [comment]
> Maybe I'm not aware if we do have an agreement on the file naming convention.
> 
>     * kebap case?
> 
>     * Camel case?
> 
> 
> My choice would be using Camel case that matches with the name of the exported component like :
> 
> https://github.com/taniarascia/takenote/tree/master/src/client/components
> 
> any thoughts?

I think this PR is not the right place to discuss it. You should open a [new discussion](https://github.com/Stooa/Stooa/discussions) for this topic

**Comment by jordisala1991** [comment]
It does not make sense to keep this PR opened.
